File: backend/src/main/java/ai/threeeye/yeyak/domain/Device.java
package ai.threeeye.yeyak.domain;

import ai.threeeye.yeyak.common.define.device.OS;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Document("devices")
public class Device {

    @Id
    @Builder.Default
    private String _id = UUID.randomUUID().toString();

    @Indexed(unique = true)
    private String code;

    private String manufacturer;
    private String model;
    private String color;
    private OS os;
    private boolean availability;
    private String description;

    @CreatedDate
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
    @LastModifiedDate
    private LocalDateTime updatedAt;


}

File: mobile/lib/models/Device.dart
class Device {
  final String id;

  final String code;

  final String manufacturer;
  final String model;
  final String color;
  final String os;
  final bool? availability;
  final String? description;

  final DateTime? createdAt;
  final DateTime? updatedAt;

  Device(
      {required this.id,
      required this.code,
      required this.manufacturer,
      required this.model,
      required this.color,
      required this.os,
      this.availability,
      this.description,
      this.createdAt,
      this.updatedAt});

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
        id: json["_id"],
        code: json["code"],
        manufacturer: json["manufacturer"],
        model: json["model"],
        color: json["color"],
        os: json["os"],
        availability: json["availability"],
        description: json["description"],
        createdAt: json["createdAt"] != null
            ? DateTime.parse(json["createdAt"])
            : null,
        updatedAt: json["updatedAt"] != null
            ? DateTime.parse(json["updatedAt"])
            : null);
  }
}
